Source/AssetRipper.Export.Modules.Shader/ShaderBlob/ShaderSubProgramBlob.cs

```diff
@@ -33,95 +33,19 @@ public void Read(AssetCollection shaderCollection, byte[] compressedBlob, AssetL
 
 		private void ReadBlob(AssetCollection shaderCollection, byte[] compressedBlob, uint offset, uint compressedLength, uint decompressedLength, int segment)
 		{
-			byte[] decompressedBuffer = new byte[decompressedLength];
-			LZ4Codec.Decode(compressedBlob, (int)offset, (int)compressedLength, decompressedBuffer, 0, (int)decompressedLength);
+			m_decompressedBlob = new byte[decompressedLength];
+			LZ4Codec.Decode(compressedBlob, (int)offset, (int)compressedLength, m_decompressedBlob, 0, (int)decompressedLength);
 
-			using MemoryStream blobMem = new MemoryStream(decompressedBuffer);
+			using MemoryStream blobMem = new MemoryStream(m_decompressedBlob);
 			using AssetReader blobReader = new AssetReader(blobMem, shaderCollection);
 			if (segment == 0)
 			{
 				Entries = ReadAssetArray(blobReader);
-				SubPrograms = CreateAndInitializeArray<ShaderSubProgram>(Entries.Length);
+				m_cachedSubPrograms.Clear();
 			}
-			ReadSegment(blobReader, segment);
```

Comment:
Is there a reason these can't be read all at once?

Reply:
It is not possible to directly determine whether an entry only contains ProgramData or Parameters here.